### PR TITLE
Report unmapped fields in integrity checks + delete Package.isWithheld and withheldReason

### PR DESCRIPTION
**Pull request by isoos**
Note: isBlocked could also be removed, but to be on the safe side I'd do that in the release after the current one.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ AppEngine version, listed here to ease deployment and troubleshooting.
 ## Next Release (replace with git tag when deployed)
  * Bump runtimeVersion to `2024.12.12`.
  * Upgraded runtime Dart SDK to `3.6.0`.
+ * Note: Started reporting unmapped fields and deleting `Package.isWithheld` and `Package.withheldReason`.
```
Comment (sigurdm) [inline CHANGELOG.md:7]:
```suggestion
 * Note: Started reporting unmapped fields 
 * Started and deleting `Package.isWithheld` and `Package.withheldReason`.
```

File: app/lib/shared/integrity.dart [hunk 2]
```diff
@@ -93,7 +98,13 @@ class IntegrityChecker {
       yield* _checkAuditLogs();
       yield* _checkModerationCases();
       yield* _reportPubspecVersionIssues();
-      // TODO: report unmapped properties
+
+      if (_unmappedFields.isNotEmpty) {
+        for (final field in _unmappedFields) {
+          if (_knownUnmappedFields.contains(field)) continue;
+        yield 'Unmapped field found: $field.';
```
Comment (sigurdm) [inline app/lib/shared/integrity.dart:105]: Can we report more? Like give some entity with this field, and the entitty type

File: app/lib/shared/integrity.dart [hunk 3]
```diff
@@ -40,6 +40,11 @@ class IntegrityChecker {
   final DatastoreDB _db;
   final int _concurrency;
 
+  static const _knownUnmappedFields = {
```
Comment (sigurdm) [inline app/lib/shared/integrity.dart:43]: Please document the scope of this (is it only about packages?)

File: app/lib/shared/integrity.dart [hunk 4]
```diff
@@ -40,6 +40,11 @@ class IntegrityChecker {
...
+    'Package.isWithheld',
+    'Package.withheldReason',
+  };
+  final _unmappedFields = <String>{};
```
Comment (sigurdm) [inline app/lib/shared/integrity.dart:47]: Just to make sure, do we create a new IntegrityChecker for each run?

File: app/lib/shared/integrity.dart (hunk 2 above)
Comment (isoos) [reply app/lib/shared/integrity.dart]: We are reporting `EntityType.fieldName` here. I'm not sure if we need to have a full entity identifier, as these fields are typically unused, but maybe we could report one id for each of the reported entries? That way we can store it in a Map and only keep the first one?

File: app/lib/shared/integrity.dart (hunk 4 above)
Comment (isoos) [reply app/lib/shared/integrity.dart:51]: Yes, this is a one-off object on each run.

File: app/lib/shared/integrity.dart (hunk 3 above)
Comment (isoos) [reply app/lib/shared/integrity.dart]: renamed + done.

File: app/lib/shared/integrity.dart (hunk 2 above)
Comment (isoos) [reply app/lib/shared/integrity.dart]: Implemented this, PTAL.

**Review by sigurdm — APPROVED** [review]

**Pull request merged**